packages/stream/src/combinators/state.ts

```diff
@@ -0,0 +1,53 @@
+import { Stream } from '@motorcycle/types'
+import { curry3 } from '167'
+import { scan } from './scan'
+import { switchMap } from './switchMap'
+
+/**
+ * Especially useful when keeping local state that also needs to be updated
+ * from an outside source.
+ * @name state<A, B>(f: (acc: A, value: B) => A, seed$: Stream<A>, values$: Stream<B>): Stream<A>
+ * @example 
+ * import { Stream } from '@motorcycle/types'
+ * import { query, dragOverEvent, dragstartEvent, dropEvent } from '@motorcycle/dom'
+ * import { sample, map, state, mapList } from '@motorcycle/stream'
+ * import { move } from '@typed/prelude'
+ * 
+ * export function ReorderableList(sources) {
+ *   const { list$, dom } = sources
+ *   const li = query('li', dom)
+ *   const dragOver$ = dragOverEvent(li)
+ *   const dragStart$ = dragstartEvent(li)
+ *   const drop$ = dropEvent(li)
+ *   const reducer$: Stream<(list: Array<string>) => Array<string>> = 
+ *     sample((to, from) => move(from, to), map(getKey, drop$), map(getKey, dragStart$))
```

Comment:
`getKey`, besides not being descriptive, isn't defined anywhere in the example, which would be fine as long as the name is descriptive.